s4e-backend/src/main/java/pl/cyfronet/s4e/listener/InvitationListener.java

```diff
@@ -0,0 +1,106 @@
+package pl.cyfronet.s4e.listener;
+
+import lombok.RequiredArgsConstructor;
+import lombok.val;
+import org.springframework.context.MessageSource;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+import org.thymeleaf.TemplateEngine;
+import org.thymeleaf.context.Context;
+import pl.cyfronet.s4e.bean.Invitation;
+import pl.cyfronet.s4e.event.OnConfirmInvitationEvent;
+import pl.cyfronet.s4e.event.OnRejectInvitationEvent;
+import pl.cyfronet.s4e.event.OnSendInvitationEvent;
+import pl.cyfronet.s4e.service.InvitationService;
+import pl.cyfronet.s4e.service.MailService;
+import pl.cyfronet.s4e.util.MailHelper;
+
+import javax.ws.rs.NotFoundException;
+import java.util.Locale;
+
+@Component
+@RequiredArgsConstructor
+public class InvitationListener {
+    private final InvitationService invitationService;
+
+    private final MessageSource messageSource;
+    private final TemplateEngine templateEngine;
+    private final MailService mailService;
+    private final MailHelper mailHelper;
+
+    @Async
+    @EventListener
+    public void handle(OnSendInvitationEvent event) {
+        val invitation = invitationService.findByToken(event.getToken(), Invitation.class)
+                .orElseThrow(() -> new NotFoundException("Invitation with your token doesn't exists"));
```

Comment:
I'd suggest to use a helper method here to generate exception, see SchemaService::constructNFE.
In this line you can just write:
```
.orElseThrow(() -> constructNFE(token));
```
BTW, correct the typo: "doesn't exists" → "doesn't exist".